prow/crier/reporters/gerrit/reporter_test.go

```diff
@@ -1504,26 +1843,159 @@ func TestReport(t *testing.T) {
 
 			reportedJobs, _, err := reporter.Report(context.Background(), logrus.NewEntry(logrus.StandardLogger()), tc.pj)
 			if err != nil {
-				t.Errorf("unexpected error: %v", err)
+				if !tc.expectError {
+					t.Errorf("Unexpected error: %v", err)
+				}
+				// if this error is expected then no need to verify anything
+				// later
+				return
 			}
 
-			if err == nil {
-				for _, include := range tc.reportInclude {
-					if !strings.Contains(fgc.reportMessage, include) {
-						t.Errorf("message: got %q, does not contain %s", fgc.reportMessage, include)
-					}
+			for _, include := range tc.reportInclude {
+				if !strings.Contains(fgc.reportMessage, include) {
+					t.Errorf("message: got %q, does not contain %s", fgc.reportMessage, include)
+				}
+			}
+			for _, exclude := range tc.reportExclude {
+				if strings.Contains(fgc.reportMessage, exclude) {
+					t.Errorf("message: got %q, unexpectedly contains %s", fgc.reportMessage, exclude)
 				}
-				for _, exclude := range tc.reportExclude {
-					if strings.Contains(fgc.reportMessage, exclude) {
-						t.Errorf("message: got %q, unexpectedly contains %s", fgc.reportMessage, exclude)
+			}
+
+			if !reflect.DeepEqual(tc.expectLabel, fgc.reportLabel) {
+				t.Errorf("labels: got %v, want %v", fgc.reportLabel, tc.expectLabel)
+			}
+			if len(reportedJobs) != tc.numExpectedReport {
+				t.Errorf("report count: got %d, want %d", len(reportedJobs), tc.numExpectedReport)
+			}
+		})
+	}
+}
+
+func TestMultipleWorks(t *testing.T) {
+	samplePJ := v1.ProwJob{
+		ObjectMeta: metav1.ObjectMeta{
+			Labels: map[string]string{
+				client.GerritRevision:    "abc",
+				kube.ProwJobTypeLabel:    presubmit,
+				client.GerritReportLabel: "same-label",
+				client.GerritPatchset:    "5",
+				kube.OrgLabel:            "same-org",
+				kube.RepoLabel:           "same-repo",
+				kube.PullLabel:           "123456",
+			},
+			Annotations: map[string]string{
+				client.GerritID:       "123-abc",
+				client.GerritInstance: "gerrit",
+			},
+			CreationTimestamp: metav1.Time{
+				Time: timeNow.Add(-time.Hour),
+			},
+		},
+		Status: v1.ProwJobStatus{
+			State: v1.FailureState,
+			URL:   "guber/bar",
+		},
+		Spec: v1.ProwJobSpec{
+			Refs: &v1.Refs{
+				Repo: "bar",
+				Pulls: []v1.Pull{
+					{
+						Number: 0,
+					},
+				},
+			},
+			Job:    "ci-bar",
+			Type:   v1.PresubmitJob,
+			Report: true,
+		},
+	}
+
+	// Running with 3 different batches to increase the chance of hitting races
+	for _, count := range []int{10, 20, 30} {
```

Comment:
FWIW Go's race detector is pretty magical and should pretty much guarantee to catch races if they're possible so this would probably work with just 2 goroutines as well. Scaling it up doesn't hurt unless we're debugging an error though so it doesn't really matter.
Thanks for adding this test.